megalodon/src/megalodon.ts

```diff
@@ -1334,13 +1334,16 @@ type Instance = {
 
 /**
  * Detect SNS type.
- * Now support Mastodon, Pleroma and Pixelfed.
+ * Now support Mastodon, Pleroma and Pixelfed. Throws an error when no known platform can be detected.
  *
  * @param url Base URL of SNS.
  * @param proxyConfig Proxy setting, or set false if don't use proxy.
  * @return SNS name.
  */
-export const detector = async (url: string, proxyConfig: ProxyConfig | false = false): Promise<'mastodon' | 'pleroma' | 'misskey'> => {
+export const detector = async (
+  url: string,
+  proxyConfig: ProxyConfig | false = false
+): Promise<'mastodon' | 'pleroma' | 'misskey' | null> => {
```

Comment:
```suggestion
): Promise<'mastodon' | 'pleroma' | 'misskey'> => {
```
This function will never return `null`, right?

Reply:
Fixed. Guess I'm not a good night worker 🙊